src/pal/src/misc/cgroup.cpp

```diff
@@ -0,0 +1,272 @@
+                {
+                    strcpy_s(cgrouppath, cgrouppathsize, cgroup_path);
+                    retval = true;
+                    break;  
+                }
+                strTok = strtok_s(NULL, ",", &context);
+            }
+        }
+    done:
+        if (line)
+            free(line);
+        if (cgroupfile)
+            fclose(cgroupfile);
+        return retval;
+    }
+
+    bool ReadMemoryValueFromFile(const char* filename, size_t* val)
+    {
+        bool result = false;
+        char *line = nullptr;
+        size_t lineLen = 0;
+        char* endptr = NULL;
+        size_t num = 0, l, multiplier;
+
+        if (val == NULL)
+            return false;
+
+        FILE* file = fopen(filename, "r");
+        if (file == NULL)
+            goto done;
+        
+        if (getline(&line, &lineLen, file) == -1)
+            goto done;
+
+        errno = 0;
+        num = strtoull(line, &endptr, 0); 
+        if (errno != 0)
+            goto done;
+
+        multiplier = 1;
+        switch(*endptr)
+        {
+            case 'g':
+            case 'G': multiplier = 1024;
+            case 'm': 
+            case 'M': multiplier = multiplier*1024;
+            case 'k':
+            case 'K': multiplier = multiplier*1024;
+        }
+
+        *val = num * multiplier;
+        result = true;
+        if (*val/multiplier != num)
+            result = false;
+    done:
+        if (file)
+            fclose(file);
+        if (line)
+            free(line);    
+        return result;
+    }
+};
+
+
+size_t
+PALAPI
+PAL_GetRestrictedPhysicalMemoryLimit()
+{
+    CGroup cgroup;
+    size_t physical_memory_limit;
+
+    if (!cgroup.GetPhysicalMemoryLimit(&physical_memory_limit))
+         physical_memory_limit = SIZE_T_MAX;
+
+    struct rlimit curr_rlimit;
+    size_t rlimit_soft_limit = (size_t)RLIM_INFINITY;
+    if (getrlimit(RLIMIT_AS, &curr_rlimit) == 0)
+    {
+        rlimit_soft_limit = curr_rlimit.rlim_cur;
+    }
+    physical_memory_limit = min(physical_memory_limit, rlimit_soft_limit);
+
+    // Ensure that limit is not greater than real memory size
+    size_t pages = (size_t) sysconf(_SC_PHYS_PAGES);
+    if (pages != -1) 
+    {
+        long pageSize = sysconf(_SC_PAGE_SIZE);
+        if (pageSize != -1)
+        {
+            physical_memory_limit = min(physical_memory_limit, 
+                                        pages * pageSize);
+        }
+    }
+
+    if(physical_memory_limit == SIZE_T_MAX)
+        physical_memory_limit = 0;
+    return physical_memory_limit;
+}
+
+BOOL
+PALAPI
+PAL_GetWorkingSetSize(size_t* val)
+{
+    BOOL result = false;
+    size_t linelen;
+    char* line = NULL;
+
+    if (val == NULL)
+        return FALSE;
+
+    FILE* file = fopen(PROC_STATM_FILENAME, "r");
+    if (file != NULL && getline(&line, &linelen, file) != -1)
+    {
+        int x = sscanf_s(line, "%*s %llu %*s %*s %*s %*s %*s", val);
+        if(x==1)
```

Comment:
A nit - could you please add spaces around the `==` to unify it with the rest of the code?